go-controller/pkg/ovn/gateway_init.go

```diff
@@ -349,28 +351,14 @@ func addDistributedGWPort() error {
 	var nbctlArgs []string
 	// add a distributed gateway port to the distributed router
 	dgpName := types.RouterToSwitchPrefix + types.NodeLocalSwitch
-	if config.IPv4Mode {
-		dgpMac = util.IPAddrToHWAddr(net.ParseIP(types.V4NodeLocalDistributedGWPortIP)).String()
-	} else {
-		dgpMac = util.IPAddrToHWAddr(net.ParseIP(types.V6NodeLocalDistributedGWPortIP)).String()
-	}
+	dgpMac = util.IPAddrToHWAddr(net.ParseIP(types.V4NodeLocalDistributedGWPortIP)).String()
+
 	nbctlArgs = append(nbctlArgs,
+		"--if-exists", "lrp-del", dgpName, "--",
```

Comment:
the `--may-exist` flag doesn´t work and we have to delete it to overwrite it
```
ovn-nbctl --may-exist lrp-add ovn_cluster_router rtos-node_local_switch 0a:58:a9:fe:00:02 169.254.0.2/20 fd99::2/64
ovn-nbctl: rtos-node_local_switch: port already exists with different network
because it exists with only one ip address
```
the alternative is  to add a more complicated logic to deal with the conversion, but since the `initJoinLogicalSwitchIPManager` if we can make it ALWAYS dual stack it simplifies everything a lot